crates/npmrc/src/custom_deserializer.rs

```diff
@@ -17,12 +17,12 @@ pub fn default_public_hoist_pattern() -> Vec<String> {
 }
 
 // Get the drive letter from a path on Windows. If it's not a Windows path, return None.
-fn get_drive_letter(current_dir: &Path) -> Option<String> {
-    for component in current_dir.components() {
-        if let Component::Prefix(prefix_component) = component {
-            return Some(
-                prefix_component.as_os_str().to_str().unwrap().replace(':', "").to_string(),
-            );
+fn get_drive_letter(current_dir: &Path) -> Option<Cow<'_, str>> {
+    if let Some(Component::Prefix(prefix_component)) = current_dir.components().next() {
+        if let std::path::Prefix::Disk(disk_byte) | std::path::Prefix::VerbatimDisk(disk_byte) =
+            prefix_component.kind()
+        {
+            return Some(Cow::Owned((disk_byte as char).to_string()));
```

Comment:
Actually, why return `Option<Cow<str>>` in the first place? Why not just return `Option<char>`? Drive letters don't need a whole string.